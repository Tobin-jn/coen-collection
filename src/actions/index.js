export const loadMovies = movies => ({
  type: "LOAD_MOVIES",
  movies
});

export const logIn = loggedIn => ({
  type: "TOGGLE_LOGIN",
  loggedIn: false
});
