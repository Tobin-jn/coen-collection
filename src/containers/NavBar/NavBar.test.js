import React from 'react';
import { mapStateToProps, mapDispatchToProps, NavBar } from './index';
import { shallow } from 'enzyme';
import { NavLink } from "react-router-dom";
import { logIn } from "../../actions";

describe('NavBar', () => {
  let wrapper;
  const mockHandleLogin = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<NavBar 
        loggedIn={true}
        currentUser={'Taylor'}
        handleLogin={mockHandleLogin} />);
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

	it("render like snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render navlinks if user is not logged in', () => {
    wrapper = shallow(<NavBar 
        loggedIn={false}
        currentUser={''}
        handleLogin={mockHandleLogin} />);
    expect(wrapper.find(NavLink).length).toEqual(2);
  });

  it('should render a signout button if user is logged in', () => {
    expect(wrapper.find(NavLink).length).toEqual(0);
    expect(wrapper.find('.sign-out-button')).toBeDefined();
  });

  it('should call handleLogin on click', () => {
    wrapper.find('.sign-out-button').simulate('click');
    expect(mockHandleLogin).toHaveBeenCalled();
  });
});

describe('mapStateToProps', () => {
	it('should return an object with a loggedIn status', () => {
		const mockState = {
			loggedIn: true
		};

		const expected = {
			loggedIn: true
		};

		const mappedProps = mapStateToProps(mockState);
		expect(mappedProps).toEqual(expected);
	});
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with a logIn action when handleLogin is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = logIn(true);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleLogin(true);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
})
