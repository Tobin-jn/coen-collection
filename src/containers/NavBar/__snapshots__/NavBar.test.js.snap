// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Renders like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    loggedIn={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Hello 
        </h1>,
        <button
          className="sign-out-button buttons"
          onClick={[Function]}
        >
          Sign Out
        </button>,
      ],
      "className": "navBar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Hello ",
          undefined,
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Out",
          "className": "sign-out-button buttons",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Sign Out",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Hello 
          </h1>,
          <button
            className="sign-out-button buttons"
            onClick={[Function]}
          >
            Sign Out
          </button>,
        ],
        "className": "navBar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hello ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Hello ",
            undefined,
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Out",
            "className": "sign-out-button buttons",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
